copystructure.go

```diff
@@ -48,14 +49,25 @@ type walker struct {
 	vals        []reflect.Value
 	cs          []reflect.Value
 	ps          []bool
+	locks       []sync.Locker
 }
 
 func (w *walker) Enter(l reflectwalk.Location) error {
 	w.depth++
+	// this normally appends only once, but make sure we start with enough
+	// space to use depth as an index
+	for w.depth >= len(w.locks) {
+		w.locks = append(w.locks, nil)
+	}
```

Comment:
Oh, not sure if my comment may have thrown you off. I wrote that in an iteration where `w.locks` was being resliced to pop off values, but now it's never truncated. This only needs to ensure it's long enough to index with `w.depth` at any point.